src/DonationForms/Actions/AddHoneyPotFieldToDonationForms.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Give\DonationForms\Actions;
+
+use Give\DonationForms\Rules\HoneyPotRule;
+use Give\Framework\FieldsAPI\DonationForm;
+use Give\Framework\FieldsAPI\Exceptions\EmptyNameException;
+use Give\Framework\FieldsAPI\Honeypot;
+
+/**
+ * @unreleased
+ */
+class AddHoneyPotFieldToDonationForms
+{
+    /**
+     * @unreleased
+     * @throws EmptyNameException
+     */
+    public function __invoke(DonationForm $form): void
+    {
+        $formNodes = $form->all();
+        $lastSection = $form->count() ? $formNodes[$form->count() - 1] : null;
+
+        if ($lastSection) {
```

Comment:
Wouldn't there always be a last section, even if there is only one section?

Reply:
Ideally yes, however if for some reason there are no sections i'm preventing that with the check for `$form->count()`

Reply:
Ok. We would probably have other problems if there are _no sections_, but let's at least not make things worse here.